packages/@aws-cdk/pipelines/lib/legacy/actions/publish-assets-action.ts

```diff
@@ -155,6 +164,8 @@ export class PublishAssetsAction extends Construct implements codepipeline.IActi
 
     this.action = new codepipeline_actions.CodeBuildAction({
       actionName: props.actionName,
+      // Add namespace to all asset steps that allow to use values from those if exported
+      variablesNamespace: `${this.stageName}@${props.actionName}`,
```

Comment:
`namespaceName()` replaces certain characters who's replacement would also be necessary here since you're allowing for user input for the `stageName`.